beacon-chain/core/helpers/validators.go

```diff
@@ -244,6 +244,9 @@ func ComputeProposerIndex(validators []*ethpb.Validator, activeIndices []uint64,
 //    fork_version = state.fork.previous_version if epoch < state.fork.epoch else state.fork.current_version
 //    return bls_domain(domain_type, fork_version)
 func Domain(fork *pb.Fork, epoch uint64, domainType []byte) uint64 {
+	if fork == nil || domainType == nil {
+		return 0
```

Comment:
0 is a valid domain no? shouldn't we return a max int in this case?

Reply:
maybe i will add error to the function?

Reply:
i think that's better than 0

Reply:
added error here, should i do the same for ActivationEligibilityEpoch? quite a lot of changes

Reply:
in ActivationEligibilityEpoch i returned far future epoch